src/main/java/io/github/christophermanahan/carnitas/Main.java

```diff
@@ -6,8 +6,8 @@
 public class Main {
 
   public static void main(String[] args) throws IOException {
-    int port = Integer.parseInt(args[0]);
+    int port = args.length == 0  ? 33333 : Integer.parseInt(args[0]);
```

Comment:
Ah right sorry I think you were using a custom `ServerSocket`

Reply:
I think there is an opportunity to use `Optional` here @christophermanahan. I think something along these lines might work.

```java
Optional.of(args)
    .filter(a -> a.length > 0)
    .map(a -> a[0])
    .map(Integer::parseInt)
    .orElse(3333);
```

It would be good to weigh whether or not this implementation more valuable than the way it is currently implemented. I will leave that up to you.

Reply:
@damonkelley I will leave the ternary for now and come back to this and address the pros and cons of the optional vs ternary approach when I handle extracting the default port to a more appropriate location